src/lsp4clj/server.clj

```diff
@@ -160,6 +165,30 @@
       (when-let [message (async/<! server-initiated-ch)]
         (protocols.endpoint/receive-response server message)
         (recur)))
+    ;; Request processing Stage 3:
+    ;; We've received the requests in series (Stage 1). The language server has
+    ;; indicated (Stage 2) whether a request has been processed synchronously or
+    ;; should be processed in parallel. In either case, here we receive a
+    ;; dereffable response*, which can be used to realize the response's value.
+    ;; We deref the responses in parallel, in 4 threads, sending the responses
+    ;; to the output. These responses may not be output in the order they were
+    ;; received, but the language server has decided that's ok.
+    (comment
+      ;; rather than hard-coding 4, we could use the same level of parallelism
+      ;; that lsp4j was using, or rather that clojure-lsp was using by calling
+      ;; `CompletableFuture/supplyAsync`.
+      (java.util.concurrent.ForkJoinPool/getCommonPoolParallelism))
+    (async/pipeline-blocking 4 ;; allow requests to be processed in parallel
```

Comment:
I think we could change to use the `java.util.concurrent.ForkJoinPool/getCommonPoolParallelism` instead of the `4`, WDYT?

Reply:
For me this returns `7`, so I can see how that could behave faster for machines with a greater parallelism

Reply:
That seems fine. I'll make that change.